app/phone/src/main/java/dev/jdtech/jellyfin/PlayerActivity.kt

```diff
@@ -120,6 +134,75 @@ class PlayerActivity : BasePlayerActivity() {
+            when (viewModel.player) {
+                is MPVPlayer -> {
+                    if (!isControlsLocked) {
+                        exoPlayerControlView.visibility = View.GONE
+                        lockedLayout.visibility = View.VISIBLE
+                        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LOCKED
+                        isControlsLocked = true
+                    } else {
+                        exoPlayerControlView.visibility = View.VISIBLE
+                        lockedLayout.visibility = View.GONE
+                        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_SENSOR_LANDSCAPE
+                        isControlsLocked = false
+                    }
+                }
+
+                is ExoPlayer -> {
+                    if (!isControlsLocked) {
+                        exoPlayerControlView.visibility = View.GONE
+                        lockedLayout.visibility = View.VISIBLE
+                        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LOCKED
+                        isControlsLocked = true
+                    } else {
+                        exoPlayerControlView.visibility = View.VISIBLE
+                        lockedLayout.visibility = View.GONE
+                        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_SENSOR_LANDSCAPE
+                        isControlsLocked = false
+                    }
+                }
+            }
+        }
+
+        unlockButton.setOnClickListener {
+            when (viewModel.player) {
+                is MPVPlayer -> {
+                    if (isControlsLocked) {
+                        exoPlayerControlView.visibility = View.VISIBLE
+                        lockedLayout.visibility = View.GONE
+                        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_SENSOR_LANDSCAPE
+                        isControlsLocked = false
+                    } else {
+                        exoPlayerControlView.visibility = View.VISIBLE
+                        lockedLayout.visibility = View.VISIBLE
+                        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LOCKED
+                        isControlsLocked = true
+                    }
+                }
+
+                is ExoPlayer -> {
+                    if (isControlsLocked) {
+                        exoPlayerControlView.visibility = View.VISIBLE
+                        lockedLayout.visibility = View.GONE
+                        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_SENSOR_LANDSCAPE
+                        isControlsLocked = false
+                    } else {
+                        exoPlayerControlView.visibility = View.GONE
+                        lockedLayout.visibility = View.VISIBLE
+                        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LOCKED
+                        isControlsLocked = true
+                    }
+                }
+            }
+        }
```

Comment:
Couldn't this whole block be replace by something much shorter:

```suggestion
        lockButton.setOnClickListener {
            exoPlayerControlView.visibility = View.GONE
            lockedLayout.visibility = View.VISIBLE
            requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LOCKED
            isControlsLocked = true
        }

        unlockButton.setOnClickListener {
            exoPlayerControlView.visibility = View.VISIBLE
            lockedLayout.visibility = View.GONE
            requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_SENSOR_LANDSCAPE
            isControlsLocked = false    
        }
```
It does not seem necessary to check what type of player is used and if the controls are locked. The lock / unlock button is only shown when unlocked / locked.

Reply:
I've been meaning to test such a change, but never got the time to properly do it.
About the type of player: I wasn't sure how you handle such things, so I added separate entries for both just to be sure.
I'll change this asap.